python/datasources/covid_tracking_project.py

```diff
@@ -14,5 +17,112 @@ def get_id():
     def get_table_name():
         return 'covid_tracking_project'
 
+    @staticmethod
+    def get_standard_columns():
+        """Returns a dict containing conversions from Covid Tracking Project's
+           race categories to their standardized values."""
+        return {
+            'aian': col_std.Race.AIAN.value,
+            'asian': col_std.Race.ASIAN.value,
+            'black': col_std.Race.BLACK.value,
+            'nhpi': col_std.Race.NHPI.value,
+            'white': col_std.Race.WHITE.value,
+            'multiracial': col_std.Race.MULTI.value,
+            'other': col_std.Race.OTHER.value,
+            'unknown': col_std.Race.UNKNOWN.value,
+            'ethnicity_hispanic': col_std.Race.HISP.value,
+            'ethnicity_nonhispanic': col_std.Race.NH.value,
+            'ethnicity_unknown': col_std.Race.ETHNICITY_UNKNOWN.value,
+            'total': col_std.Race.TOTAL.value
+        }
+
     def write_to_bq(self, dataset, gcs_bucket, **attrs):
-        logging.info("Covid Tracking Project write_to_bq not implemented yet")
+        filename = self.get_attr(attrs, 'filename')
+        metadata_table_id = self.get_attr(attrs, 'metadata_table_id')
+        table_name = self.get_attr(attrs, 'table_name')
+
+        df = gcs_to_bq_util.load_csv_as_dataframe(gcs_bucket, filename)
+        self.clean_frame_column_names(df)
+
+        # Massage the data into the standard format.
+        df.drop(columns=['cases_latinx', 'deaths_latinx',
+                         'hosp_latinx', 'tests_latinx'], inplace=True)
+        df = df.melt(id_vars=['date', 'state'])
+        df[['variable_type', 'race_and_ethnicity']] = df.variable.str.split(
+            "_", 1, expand=True)
+        df.drop('variable', axis=1, inplace=True)
+        df.rename(columns={'state': 'state_postal_abbreviation'}, inplace=True)
+        df.replace({'race_and_ethnicity': self.get_standard_columns()},
+                   inplace=True)
+
+        # Get the metadata table
+        metadata = self._download_metadata(metadata_table_id)
+        if len(metadata.index) == 0:
+            raise RuntimeError(
+                'BigQuery call to {} returned 0 rows'.format(metadata_table_id))
+
+        # Merge the tables
+        merged = pd.merge(
+            df, metadata, how='left',
+            left_on=['state_postal_abbreviation', 'variable_type'],
+            right_on=['state_postal_abbreviation', 'variable_type'])
+        # Rename combined race categories
+        self._rename_race_category(merged, 'reports_api', col_std.Race.ASIAN,
+                                   col_std.Race.API)
+        self._rename_race_category(merged, 'reports_ind', col_std.Race.AIAN,
+                                   col_std.Race.INDIGENOUS)
+
+        merged.drop(columns=['reports_api', 'reports_ind'], inplace=True)
+
+        # Write to BQ
+        gcs_to_bq_util.append_dataframe_to_bq(merged, dataset, table_name)
```

Comment:
Some general data cleanup stuff:
- Could we convert the dates to `yyyy-mm-dd` format?
- Not sure if CTP fixed this issue, but in the past when I was running a script on their data, some of the int values were actually strings with commas in them, so I had to [fix those](https://github.com/aaronsn/health-equity-tracker/blob/566860cf99fe1fd3be65ca930db1c677d9efcdbf/python/datasources/covid_tracking_project.py#L103) before uploading to BQ

Reply:
Yeah, I'll format the dates.

I haven't been able to repro the ints as strings issue you describe. The CTP data is a CSV, so the data is untyped and pandas has been importing them as ints

Reply:
Oooh okay I figured out the strings issue. I found if you set thousands=',' in the call to pd.read_csv, it will parse the numbers correctly. Woo!!

Reply:
wooooah. That's pretty neat, nice find! Of course pandas can do this automagically